src/main/java/percy/ui/Ui.java

```diff
@@ -0,0 +1,179 @@
+                    + "  |  __ \\\n"
+                    + "  | |__) |__ _ __ ___ _   _\n"
+                    + "  |  ___/ _ \\ '__/ __| | | |\n"
+                    + "  | |  |  __/ | | (__| |_| |\n"
+                    + "  |_|   \\___|_|  \\___|\\__, |\n"
+                    + "                      __/ |\n"
+                    + "                     |___/\n";
+    private static final String MESSAGE_WELCOME = "    Hello! I'm Percy\n" + "    What can I do for you?";
+    private static final String MESSAGE_GOODBYE = "Bye. Hope to see you again soon!";
+
+    /**
+     * Constructs an user interface.
+     * @param in input stream.
+     * @param out print stream.
+     */
+    public Ui(InputStream in, PrintStream out) {
+        this.in = new Scanner(in);
+        this.out = out;
+    }
+
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    /**
+     * Reads the command input by user.
+     * @return the command given.
+     */
+    public static String readCommand() {
+        String command = "";
+        while (command.trim().isEmpty()) {
+            command = in.nextLine();
+        }
+        return command;
+    }
+
+    /**
+     * Prints the start up message.
+     */
+    public static void printStartUpMsg() {
+        System.out.println(MESSAGE_LOGO + "\n" + DIVIDER
+                + MESSAGE_WELCOME + "\n"
+                + DIVIDER);
+    }
+
+    /**
+     * Makes the startup message.
+     * @return  formatted startup string.
+     */
+    public static String makeStartUpMsg() {
+        return MESSAGE_LOGO + "\n" + DIVIDER
+                + MESSAGE_WELCOME + "\n"
+                + DIVIDER;
+    }
+
+    /**
+     * Makes the goodbye message.
+     * @return formatted goodbye string.
+     */
+    public static String makeByeMsg() {
+        return MESSAGE_GOODBYE + "\n";
+    }
+
+    /**
+     * Formats any string for output purposes.
+     * @return formatted string.
+     */
+    public static String makeMsg(String s) {
+        return s.toString() + "\n";
+    }
+
+    /**
+     * Formats any array of string for multi-line output purposes.
+     * @param stringArr array of strings.
+     * @return formatted multi-line string.
+     */
+    public static String makeMsg(ArrayList<String> stringArr) {
+        String str = "";
+        for (String s : stringArr) {
+            str += s.toString() + "\n";
+        }
+        return str;
+    }
+
+    /**
+     * Formats the output message for listing of tasks.
+     * @param list latest task list.
+     * @return the output message for list command
+     */
+    public static String makeListMsg(TaskList list) {
+        String response;
+        response = "Here are the tasks in your list:\n";
+        int i = 1;
+        for (Task t : list.getTaskList()) {
+            response += String.valueOf(i++) + ". "
+                    + t.toString() + "\n";
+        }
+        String taskString = (list.getTaskList().size() == 1) ? "task" : "tasks";
```

Comment:
I like that you made an effort to distinguish between singular and plural tasks! :)